marklogic-data-hub-spark-connector/src/test/java/com/marklogic/hub/spark/sql/sources/v2/reader/ReadAllDataTypesTest.java

```diff
@@ -0,0 +1,91 @@
+package com.marklogic.hub.spark.sql.sources.v2.reader;
+
+import com.marklogic.client.io.DocumentMetadataHandle;
+import com.marklogic.client.io.Format;
+import com.marklogic.client.io.InputStreamHandle;
+import com.marklogic.hub.HubClient;
+import com.marklogic.hub.spark.sql.sources.v2.Options;
+import com.marklogic.io.Base64;
+import org.apache.commons.codec.binary.Hex;
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.types.DataTypes;
+import org.junit.jupiter.api.Test;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class ReadAllDataTypesTest extends AbstractSparkReadTest{
+
+    @Test
+    void readAllDataTypes() {
+        runAsDataHubDeveloper();
+        loadTDE("User");
+
+        HubClient client = runAsDataHubOperator();
+        InputStreamHandle handle = new InputStreamHandle(readInputStreamFromClasspath("entityInstances/User/user1.json"));
+        handle.withFormat(Format.JSON);
+        client.getFinalClient().newJSONDocumentManager().write("/doc1.json", new DocumentMetadataHandle()
+                .withCollections("User")
+                .withPermission("data-hub-operator", DocumentMetadataHandle.Capability.READ, DocumentMetadataHandle.Capability.UPDATE),
+            handle);
+
+
+        Options options = newOptions().withView("User").withNumPartitions("2");
+        HubDataSourceReader dataSourceReader = new HubDataSourceReader(options.toDataSourceOptions());
+        List<InternalRow> rows = readRows(dataSourceReader);
+        assertEquals(1, rows.size());
+        for (InternalRow row : rows) {
+            //The comment in the assertions will have the datatype of the property in TDE template
+            assertEquals(1, row.getInt(0), "integer");
+            assertEquals(1, row.getInt(1), "int");
+            assertEquals(Byte.decode("1"), row.getByte(2),"byte");
+            assertEquals(1, row.getInt(3),"positiveInteger");
+            assertEquals(1, row.getInt(4), "nonNegativeInteger");
+            assertEquals(1L, row.getLong(5), "long");
+            assertEquals(1L, row.getLong(6), "unsignedLong");
+            assertEquals(1L, row.getLong(7), "unsignedInt");
+            assertEquals(Short.decode("1"), row.getShort(8), "short");
+            assertEquals(Short.decode("1"), row.getShort(9), "unsignedByte");
+            assertEquals(Short.decode("1"), row.getShort(10), "unsignedShort");
+            assertEquals("John", row.getString(11), "string");
+            assertEquals(6218, row.get(12, DataTypes.DateType), "date");
+            assertEquals(537336758000000L, row.get(13, DataTypes.TimestampType), "dateTime");
```

Comment:
Both this and the date row should have assertion messages explaining why these are the expected values, as it's not clear to the casual reader